src/libs/Violations/ViolationsUtils.ts

```diff
@@ -100,15 +110,40 @@ function getTagViolationsForMultiLevelTags(
                 break;
             }
         }
+
         if (!hasInvalidTag) {
             newTransactionViolations = reject(newTransactionViolations, {
                 name: CONST.VIOLATIONS.TAG_OUT_OF_POLICY,
             });
         }
     }
+
     return newTransactionViolations;
 }
 
+/**
+ * Calculates some tag levels required and missing tag violations for the given transaction
+ */
+function getTagViolationsForMultiLevelTags(
+    updatedTransaction: Transaction,
+    transactionViolations: TransactionViolation[],
+    policyRequiresTags: boolean,
+    policyTagList: PolicyTagList,
+    hasDependentTags: boolean,
+): TransactionViolation[] {
+    const policyTagKeys = getSortedTagKeys(policyTagList);
+    const selectedTags = updatedTransaction.tag?.split(CONST.COLON) ?? [];
+    const filteredTransactionViolations = transactionViolations.filter(
+        (violation) => violation.name !== CONST.VIOLATIONS.SOME_TAG_LEVELS_REQUIRED && violation.name !== CONST.VIOLATIONS.TAG_OUT_OF_POLICY,
```

Comment:
Let's remove `CONST.VIOLATIONS.MISSING_TAG` from here as well. The idea is that we first clear all the tag-related violations, and then we calculate them again